npm-packages.md

```diff
@@ -1,14 +1,117 @@
-# Publishing and using `@guardian` NPM packages
+# NPM packages
 
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## Using
+## Publishing a `@guardian` library as an NPM package
 
--   Make sure you include any `@guardian` packages for transpilation in your build process
-    -   e.g. your [`webpack.config.js`](https://github.com/webpack/webpack/issues/2031#issuecomment-219040479) might including something that looks like this:
+### Authoring
+
+Write your library in [TypeScript](https://www.typescriptlang.org).
+
+### Packaging
+
+Do not include your TypeScript source code in your package. Instead, compile your source to JavaScript and [declaration files](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html) and publish those.
+
+> TypeScript depends on a project-specific configuration, and your setup may not be the same the project that installs your package
+
+Do not include or depend on [polyfills](https://developer.mozilla.org/en-US/docs/Glossary/Polyfill). Let projects that install your package decide what they need (see the section on using our packages below).
```

Comment:
I wonder if this needs fleshing out or defining further. I guess whether a polyfill is needed depends on the target browsers that a project supports. So we're rightly saying that projects, not libraries, should decide which polyfills to bring.

But what does it mean to say to library authors "Do not depend on polyfills"? Are we saying "Do not depend on language features that require polyfills as of ES2020?"

Let me know if what I'm saying makes no sense!

Reply:
ah right, no i meant to say that you can write anything that TS will let you when you target ES2020, and applications will manage making the code you export work.

does it need rewording?

Reply:
I think so? I'll have a go at it and see what people think